core/domain/takeout_service_test.py

```diff
@@ -607,3 +621,39 @@ def test_export_data_nontrivial(self):
 
         exported_data = takeout_service.export_data_for_user(self.USER_ID_1)
         self.assertEqual(exported_data, expected_export)
+
+
+    def test_get_models_to_export(self):
+        """Ensure that the set of models to export is the set of models with
+        export policy CONTAINS_USER_DATA, and that all other models have
+        export policy NOT_APPLICABLE.
+        """
+        all_models = []
+        model_names_list = [
+            model_name
+            for model_name in dir(models.NAMES)
```

Comment:
Don't put logic in tests. Find a different way to verify the output of the method, otherwise there's no point in the test -- you're just repeating the same implementation twice.

Reply:
Isn't this different logic? This is just a way for getting all of the models. I need all the models in both cases, but the difference here is that I verify the export policies of the models rather than actually calling export. Otherwise, if I need to make a change here, I would want to change the set of strings returned in the get_models_should_be_exported function to somehow return models that I can directly call export on, but I'm not sure how this can be done.

Reply:
Ah, I made a mistake in the previous comment. Sorry for the confusion.

The part that's in code should be an explicit list (see other comment). The test can do what it's doing now (though maybe you can find a way to simplify all the string parsing stuff, I'm not sure it's actually needed).

Reply:
Resolved in latest commit.